### ISSUE TITLE
Do not show SC2059 warning if printf has multiple non-option arguments

### ISSUE THREAD

**Issue by marveloo** [koalaman/shellcheck#3230]
When `printf` is used with more than one non-option argument, it strongly suggests that the first argument is intended to be used as a format string. Otherwise, the presence of additional arguments would not be meaningful. Therefore, it seems reasonable to suppress the [SC2059](https://www.shellcheck.net/wiki/SC2059) warning in cases where multiple non-option arguments are passed to `printf`.

```bash
# Show SC2059 warning for this:
printf "$pattern"

# However, do not show a warning for these:
printf "$pattern" "$value"
printf "$pattern" "$value1" "$value2" "$value3"
```
